src/model/operation/batchnorm.cc

```diff
@@ -17,8 +17,187 @@ BatchNormHandle::BatchNormHandle(const float momentum, const Tensor& input) {
   } else {
     LOG(FATAL) << "The dimension of input should either be 4D or 2D.";
   }
+
+
+#ifdef USE_MKLDNN
+  dtype = GetMKLDNNDataType(input.data_type());
+  epsilon =1e-5f;
+  data_memory_format = is_2d ? mkldnn::memory::format::nc : mkldnn::memory::format::nchw;
+  if (is_2d) {
+    x_dims = {(int)batchsize, (int)channels};
+    y_dims = {(int)batchsize, (int)channels};
+  } else {
+    x_dims = {(int)batchsize, (int)channels, (int)height, (int)width};
+    y_dims = {(int)batchsize, (int)channels, (int)height, (int)width};
+  }
+
+  auto eng = *input.device()->context(0)->engine;
+  x_md = new mkldnn::memory::desc(x_dims, dtype, data_memory_format);
+  dx_md = new mkldnn::memory::desc(x_dims, dtype, data_memory_format);
+  bn_fwd_d = new mkldnn::batch_normalization_forward::desc(mkldnn::forward_training, *x_md, epsilon,
+                                                           mkldnn::use_scale_shift);
+  bn_fwd_pd = new mkldnn::batch_normalization_forward::primitive_desc(*bn_fwd_d, eng);
+#endif // USE_MKLDNN
+
 };
 
+
+  BatchNormHandle::~BatchNormHandle() {
+#ifdef USE_MKLDNN
+    delete (x_md);
+    delete (dx_md);
+    delete (bn_fwd_d);
+    delete (bn_fwd_pd);
+#endif // USE_MKLDNN
+  }
+
+#ifdef USE_MKLDNN
+
+  Tensor CpuBatchNormForwardInference(const BatchNormHandle &bnh, const Tensor& x, const Tensor& bnScale, const Tensor& bnBias,
+                                                        Tensor& running_mean, Tensor& running_var){
```

Comment:
running_mean and running_var are not used?

Reply:
fixed and updated